contracts/modules/SuperMinterV1_1.sol

```diff
@@ -319,45 +330,60 @@ contract SuperMinterV1_1 is ISuperMinterV1_1, EIP712 {
 
         _incrementMinted(mode, d, p);
 
-        /* ----------- AFFILIATE AND PLATFORM FEES LOGIC ------------ */
+        /* ----------------- COMPUTE AND ACCRUE FEES ---------------- */
 
         TotalPriceAndFees memory f = _totalPriceAndFees(p.tier, d, p.quantity, p.signedPrice);
-
-        uint256 remaining; // The fee sent to the Sound Edition (i.e. artist fee).
-        bool affiliated;
+        MintedLogData memory l;
 
         unchecked {
             if (msg.value != f.total) revert WrongPayment(msg.value, f.total); // Require exact payment.
 
-            remaining = f.total - f.platformFee; // `platformFee <= total`;
-            platformFeesAccrued[d.platform] += f.platformFee; // Accrue the platform fee.
+            l.finalArtistFee = f.total - f.platformFee; // `platformFee <= total`;
+            l.finalPlatformFee = f.platformFee; // Initialize to the platform fee.
+            l.affiliate = p.to == p.affiliate ? address(0) : p.affiliate; // Yeah, we know it's left curved.
 
-            if (affiliated = _isAffiliatedWithProof(d, p.affiliate, p.affiliateProof)) {
-                remaining -= f.affiliateFee; // `affiliateFee <= remaining`.
-                affiliateFeesAccrued[p.affiliate] += f.affiliateFee; // Accrue the affiliate fee.
+            /* --------------------- AFFILIATE FEES --------------------- */
+
+            if (l.affiliated = _isAffiliatedWithProof(d, l.affiliate, p.affiliateProof)) {
+                l.finalArtistFee -= f.affiliateFee;
+                l.finalPlatformFee -= f.affiliateIncentive;
+                l.finalAffiliateFee = f.affiliateFee + f.affiliateIncentive;
+                affiliateFeesAccrued[p.affiliate] += l.finalAffiliateFee;
             } else {
                 // Proof may be invalid, revert to prevent unintended skipping of affiliate fee.
                 if (p.affiliate != address(0)) revert InvalidAffiliate();
-                f.affiliateFee = 0; // Set the affiliate fee to zero if not affiliated.
             }
+
+            /* --------------------- FREE MINT FEES --------------------- */
+
+            if (f.freeMintIncentive != 0 && f.unitPrice == 0) {
+                l.finalPlatformFee -= f.freeMintIncentive;
+                l.finalFreeMintFee = f.freeMintIncentive;
+                l.finalArtistFee += l.finalFreeMintFee;
+            }
+
+            /* ------------------ FIRST COLLECTOR FEES ------------------ */
+
+            if (firstCollector[p.edition] == address(0)) firstCollector[p.edition] = p.to;
```

Comment:
I see this as most valuable outside for rewarding the first collector bringing the contract on chain since they pay extra fees. 

If the first minter isn't bringing the contract on chain and not paying any extra fees for that, this is a random reward for being first and a target for frontrunning. Lmk if you have other thoughts here